app/assets/javascripts/discourse/components/text-field.js.es6

```diff
@@ -1,7 +1,35 @@
 import computed from "ember-addons/ember-computed-decorators";
+import { siteDir } from "discourse/lib/text-direction";
+import { isRTL } from "discourse/lib/text-direction";
+import { isLTR } from "discourse/lib/text-direction";
 
 export default Ember.TextField.extend({
-  attributeBindings: ['autocorrect', 'autocapitalize', 'autofocus', 'maxLength'],
+  attributeBindings: ['autocorrect', 'autocapitalize', 'autofocus', 'maxLength', 'dir'],
+
+  @computed
+  dir() {
+    if (Discourse.SiteSettings.support_mixed_text_direction) {
```

Comment:
In a component you should use `this.siteSettings` instead of `Discourse.SiteSettings`